### PR TITLE
Send server logs to syslog on platforms that support it

### LINKED ISSUE #156: node should store log messages emitted by running servers
Currently, output emitted by node-launched servers (e.g. stuff they write to stdout and stderr) gets captured by the node in a rolling buffer so you can look at the last few lines of output, but otherwise it just goes into the ether. We should set things up to actually save this output so that we can use it for debugging issues, etc.

### PR DESCRIPTION
**Pull request by ejeffrey**
On linux and OSX, servers started by the node will log their stdout and stderr to the system syslog facility.  Partially fixes #156.  A windows solution would be nice.

### CODE REVIEW

**Comment by maffoo** [comment]
I think we should set up the servers to do the logging themselves; that way we get logs whether you start a server by running it directly or by launching it from the node. One might argue that we usually launch servers from the node, but, well, we also usually run servers on windows :)

Speaking of windows, since we want something which will work cross-platform with minimal setup, I wonder if we ought to just do direct file logging instead. We could for example add an environment variable like `LABRAD_LOG_DIR` and then set up python/twisted logging to put logs for a server in there; for example server foo would log to `$LABRAD_LOG_DIR/foo/foo.log` (put it in a subdirectory because the log files will get rotated and it will keep things cleaner). The log directory could default to `~/.labrad/logs` (or maybe `/var/log/labrad` on unix-y systems).

Thoughts?

**Comment by ejeffrey** [comment]
My thinking was that when we run servers from the command line we want the
log to go to stdout anyway.  Also, putting it in the node makes it easy to
add new logging options and keep the configuration in one place (for
instance, update the node to do direct file logging) and all the servers
automatically get that, including java/scala servers.

**Comment by maffoo** [comment]
That makes sense; it is nice to just print to std out in servers, and let the system handle capturing that in log files.

A few questions about syslog; in a default configuration (on ubuntu, say), where will the logs end up? Will they be in a single file for the node, or in some system file with other user programs too? Is it possible to have the server name function as the prefix for messages from a given server, without also having labrad-node as a prefix for all those messages? It be nice to be able to distinguish between messages coming directly from servers, and messages coming from the node about the servers (e.g. when they are started and stopped).

**Comment by ejeffrey** [comment]
The default is to put everything in /var/log/syslog (on ubuntu: OSX is different).  However, you can set filters that put all the labrad messages in a single file, or each server in its own file.  I can also have the service be the server that generated the message rather than Labrad-Node.

The things l like about syslog is that it automatically handles log rotation, compression, and concurrent access, stores the logs in the correct place for the platform, and most importantly can be run without having write access to the filesystem so you can run servers with a locked down user account with no direct filesystem write access.

I am updating this branch to have node configuration options to select between a syslog and file backend.

**Comment by maffoo** [comment]
How would that work to use the server name as the identity of the log message? Would you just have to call `syslog.openlog` before every `syslog.syslog` call? It's too bad that this package seems to just use globals...

**Comment by ejeffrey** [comment]
Yeah, it just uses the posix syslog library.  There is also a syslog
implementation in the python logging module.  I will check that out.

**Comment by ejeffrey** [comment]
This update supports both file and syslog backends, and logs to a service name that looks like labrad.data_vault (to enable easy log searching)

File: labrad/node/__init__.py [hunk 1]
```diff
@@ -85,8 +86,9 @@
 from twisted.python import log, failure, usage
 from twisted.python.components import registerAdapter
 from twisted.plugin import getPlugins
-
 from zope.interface import Interface, implements
+import logging
+import logging.handlers
```
Comment (maffoo) [inline labrad/node/__init__.py]: These should go up in the first section with the standard-library imports.

**Comment by ejeffrey** [comment]
whitespace and import location fixed.

File: labrad/node/__init__.py [hunk 2]
```diff
@@ -61,6 +61,14 @@
...
 """
 
+###
+#  Syslog can be configured to filter messages generated by the node and its servers.
+#  For rsyslogd on ubuntu, create the following file:
+#  /etc/rsyslog.d/71-labrad.conf
+#
+#  :syslogtag,contains,"labrad"			/var/log/labrad.log;RSYSLOG_TraditionalFileFormat
+#
+###
```
Comment (maffoo) [inline labrad/node/__init__.py]: let's just include this comment with the module docstring immediately above.

**Comment by maffoo** [comment]
One comment about moving the rsyslog config info into the module docstring. Otherwise, LGTM

**Pull request merged**